controllers/storagecluster/cephcluster.go

```diff
@@ -800,6 +804,62 @@ func newStorageClassDeviceSets(sc *ocsv1.StorageCluster, serverVersion *version.
 		}
 	}
 
+	if sc.Spec.ManagedResources.CephNonResilientPools.Enable {
+		// Creating osds for non-resilient pools
+		for _, failureDomainValue := range sc.Status.FailureDomainValues {
+			ds := rookCephv1.StorageClassDeviceSet{}
+			ds.Name = failureDomainValue
+			ds.Count = 1
+			ds.Resources = defaults.DaemonResources["osd"]
+			// passing on existing defaults from existing devcicesets
+			ds.TuneSlowDeviceClass = sc.Spec.StorageDeviceSets[0].Config.TuneSlowDeviceClass
+			ds.TuneFastDeviceClass = sc.Spec.StorageDeviceSets[0].Config.TuneFastDeviceClass
+			annotations := map[string]string{
+				"crushDeviceClass": failureDomainValue,
+			}
+			// using the spec for volumeclaimtemplate from existing devicesets including the storageclass
+			volumeClaimTemplateSpec := storageClassDeviceSets[0].VolumeClaimTemplates[0].Spec
```

Comment:
We really should allow for setting a different volumeClaimTemplate than what is being used by the replica pools. Local storage could have different properties, such as storage class from which to provision the storage, or different size of the storage. This support can come in a future PR after we confirm the requirement with PM, but I would think it's a requirement for GA.